src/main.cpp

```diff
@@ -6087,6 +6100,14 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
         CXRequestThinBlockTx thinRequestBlockTx;
         vRecv >> thinRequestBlockTx;
 
+        // Message consistency checking
+        if (thinRequestBlockTx.setCheapHashesToRequest.empty() || thinRequestBlockTx.blockhash.IsNull())
+        {
+            LOCK(cs_main);
+            Misbehaving(pfrom->GetId(), 100);
+            return error("incorrectly constructed get_xblocktx received.  Banning peer=%d\n", pfrom->id);
```

Comment:
good one..they used to be logprints, that's they got in there..thanks.  (same for all )